### PR TITLE
Feature/four 12691: Performance: Implement the Infinite scroll

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps
image-20231204-150121

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-12691

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next
ci:deploy

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-39329186e2.eng.processmaker.net

File: resources/js/processes-catalogue/components/ProcessesCatalogue.vue [hunk 1]
```diff
@@ -62,15 +63,20 @@ export default {
...
   },
   methods: {
+    addCategories() {
+      this.numCategories += 15;
+      this.getCategories();
+    },
     getCategories() {
       ProcessMaker.apiClient
-        .get("process_categories")
+        .get(`process_categories?per_page=${this.numCategories}`)
```
Comment (fagubla) [inline resources/js/processes-catalogue/components/ProcessesCatalogue.vue:79]: improve this call, 
In the array listCategories has to be added in increments of 15.

**Review by fagubla — CHANGES_REQUESTED** [review]

File: resources/js/processes-catalogue/components/ProcessesCatalogue.vue [hunk 2]
```diff
@@ -62,17 +63,26 @@ export default {
...
+      this.page += 1;
+      this.getCategories();
+    },
     getCategories() {
       ProcessMaker.apiClient
-        .get("process_categories")
+        .get(`process_categories?page=${this.page}&per_page=${this.numCategories}`)
         .then((response) => {
-          this.listCategories = response.data.data;
+          const newCategories = response.data.data;
```
Comment (fagubla) [inline resources/js/processes-catalogue/components/ProcessesCatalogue.vue:82]: Use the spread operator, like this
`this.listCategories = [...this.listCategories, ...response.data.data]`

**Review by fagubla — CHANGES_REQUESTED** [review]

**Review by fagubla — APPROVED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-39329186e2.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-39329186e2.eng.processmaker.net

**Pull request merged**